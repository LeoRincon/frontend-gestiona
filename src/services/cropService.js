import { API_URL, CROPS_PATH } from "../utils/const";

export async function fetchCrops() {
  const sessionUser = sessionStorage.getItem("user_data");

  if (!sessionUser) {
    console.log("Invalid project ID"); // volver al user a la vista de projects
  }

  const { projectId } = JSON.parse(sessionUser);
  const url = `${API_URL}${CROPS_PATH}/project/${projectId}`;

  const response = await fetch(url, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });

  const { crops } = await response.json();

  const newData = crops.reduce((acumulador, crop, index) => {
    const { nombre, tipo_siembra, fecha_inicio, area_terreno, unit } = crop;

    const newCrop = {
      id: index + 1,
      nombre,
      tipo_siembra,
      fecha_inicio,
      area_terreno,
      unidad: unit.unidad,
      descripcion: unit.nombre,
    };
    acumulador.push(newCrop);
    return acumulador;
  }, []);

  return newData;
}

export async function createCrop(data) {
  const sessionUser = sessionStorage.getItem("user_data");

  if (!sessionUser) throw new Error("Invalid session");

  const { projectId } = JSON.parse(sessionUser);

  if (!projectId) throw new Error("Invalid Project ID");
  if (!data || typeof data !== "object") throw new Error("Data is required");

  data["proyecto_id"] = projectId;
  data["area_terreno"] = Number(data.area_terreno);

  const url = API_URL + CROPS_PATH;

  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  });

  if (!response.ok) throw new Error("Error creating Crop at service");

  const { crop } = await response.json();
  const { nombre, tipo_siembra, fecha_inicio, area_terreno, unit } = crop;

  return {
    nombre,
    tipo_siembra,
    fecha_inicio,
    area_terreno,
    unidad: unit.unidad,
    descripcion: unit.nombre,
  };
}

export async function deleteCropByName(nameCrop) {
  if (!nameCrop) throw new Error("ID Crop is required");
  if (typeof nameCrop !== "string") throw new Error("ID Crop must be a string");

  const queryUrl = API_URL + "/query";

  const query = {
    key: "nombre",
    value: nameCrop,
    returnValue: "id",
    table: "cultivo",
  };

  const queryRes = await fetch(queryUrl, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(query),
  });

  if (!queryRes.ok) throw new Error("Error Fetching Id");

  const queryData = await queryRes.json();

  const idCrop = queryData.data.id;

  const url = `${API_URL}${CROPS_PATH}/${idCrop}`;

  const response = await fetch(url, {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json",
    },
  });

  if (!response.ok) throw new Error("Error Deleting crop");

  return {
    success: true,
  };
}
