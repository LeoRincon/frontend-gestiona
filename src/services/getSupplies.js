export async function fetchSupplies() {
  const url = "http://localhost:3000/api/v1/supplies"

  const response = await fetch(url,{
    method: "GET",
    headers: {
      "Content-Type": "application/json"
    }
  })

  const {data} = await response.json()
  const newData =data.map((supply) => {
    return {
      id: supply.id,
      nombre: supply.nombre,
      cantidad_disponible: supply.cantidad_disponible,
      fecha_ingreso: supply.fecha_ingreso,
      precio: supply.precio,
      id_inventario: supply.id_inventario,
      id_categoria: supply.id_categoria,
      id_unidad_medida: supply.id_unidad_medida,
    }
  })
  return newData
}

// "id": "b6789012-8def-1234-5678-abc456789abc",
//       "nombre": "Semillas de Ma√≠z",
//       "cantidad_disponible": 200,
//       "fecha_ingreso": "2024-01-01T05:00:00.000Z",
//       "precio": 50,
//       "id_inventario": "a5678901-8abc-1234-5678-abc456789abc",
//       "id_categoria": "e3456789-8abc-1234-5678-abc456789def",
//       "id_unidad_medida": "e2345678-9def-1234-5678-abc456789012"
//     }