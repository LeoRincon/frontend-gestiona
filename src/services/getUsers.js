export async function fetchUsers() {
  const url = "http://localhost:3000/api/v1/users"
  
  const response = await fetch(url, {
    method: "GET",
    headers: {
      "Content-Type": "application/json"
    }
  })

  const { data } = await response.json()

  const newdata = data.map((user) => {
    return {
      name: user.nombre,
      email: user.email,
    }
  })
  return newdata
}