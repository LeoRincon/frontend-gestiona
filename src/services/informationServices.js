export async function getInformation() {
  const url = "http://localhost:3000/api/v1/expenses";
  const urlSales = "http://localhost:3000/api/v1/sales";
  const urlProduction = "http://localhost:3000/api/v1/product";
  const urlSupplies = "http://localhost:3000/api/v1/supplies";
  const urlActivitiesManagement = "http://localhost:3000/api/v1/activity_management";

  const response = await fetch(url, { method: "GET", headers: { "Content-Type": "application/json" } });
  const data = await response.json();

  const responseVentas = await fetch(urlSales, { method: "GET", headers: { "Content-Type": "application/json" } });
  const salesData = await responseVentas.json();

  const responseProduccion = await fetch(urlProduction, { method: "GET", headers: { "Content-Type": "application/json" } });
  const productionData  = await responseProduccion.json();

  const responseInsumos = await fetch(urlSupplies, { method: "GET", headers: { "Content-Type": "application/json" } });
  const suppliesData = await responseInsumos.json();

  const responseGestionActividades = await fetch(urlActivitiesManagement, { method: "GET", headers: { "Content-Type": "application/json" } });
  const activityManagementData = await responseGestionActividades.json();

  return {
      expenses: data,
      sales: salesData,
      production: productionData,
      supplies: suppliesData,
      activityManagements: activityManagementData,
  };
}