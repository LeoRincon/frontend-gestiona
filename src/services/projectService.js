import { API_URL, PROJECTS_PATH } from "../utils/const";

export async function fetchProjects() {
  const url = API_URL + PROJECTS_PATH;

  const response = await fetch(url, {
    method: "GET",
    headers:{
      "Content-Type": "application/json"
    }
  })

  const data = await response.json()
  
  return data
}

export async function createProject(project) {
  const url = API_URL + PROJECTS_PATH;

  if (!project || typeof project !== "object")
    throw new Error("The project must be an object");
  
  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(project),
  });

  const data = await response.json();
  return data;
}
