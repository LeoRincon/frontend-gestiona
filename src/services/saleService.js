import { API_URL, SALES_PATH, SEASONS_PATH, UNITS_PATH } from "../utils/const";

const salesUrl = API_URL + SALES_PATH;
const seasonsUrl = API_URL + SEASONS_PATH;
const unitsUrl = API_URL + UNITS_PATH;
const urls = [salesUrl, seasonsUrl, unitsUrl];

export async function createSale(sale) {
  try {
    const response = await fetch(salesUrl, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(sale),
    });
    if (!response.ok) throw new Error("Error creating sale");
    const data = await response.json();

    if (!data.success) throw new Error("Error creating sale");

    return data.sale;
  } catch (error) {
    console.error("Error creating sale: ", error);
  }
}

export async function getSalesData() {
  try {
    const [salesRes, seasonsRes, unitsRes] = await Promise.all(
      urls.map((url) =>
        fetch(url, {
          method: "GET",
          headers: { "Content-Type": "application/json" },
        })
      )
    );
    if (![salesRes, seasonsRes, unitsRes].every((res) => res.ok)) {
      throw new Error("Una o mas repuestas de la API no fueron exitosas");
    }
    const [salesData, seasonsData, unitsData] = await Promise.all([
      salesRes.json(),
      seasonsRes.json(),
      unitsRes.json(),
    ]);
    const data = salesData.sale.map((sale, index) => {
      const {
        cantidad_vendida: cantidadVendida,
        fecha_venta: fechaVenta,
        id,
        id_temporada: idTemporada,
        id_unidad_medida: idUnidadMedida,
        observaciones,
        precio_total: precioTotal,
        precio_unitario: precioUnitario,
      } = sale;
      const season = seasonsData.find((season) => season.id === idTemporada);
      const unit = unitsData.units.find((unit) => unit.id === idUnidadMedida);

      return {
        id: index + 1,
        temporada: season.nombre,
        fechaVenta,
        cantidadVendida,
        unidadMedida: unit.nombre,
        precioUnitario,
        precioTotal,
        observaciones,
        idSale: id,
        idTemporada,
      };
    });
    return data;
  } catch (error) {
    console.error("error fetching data", error);
    return (data = []);
  }
}
