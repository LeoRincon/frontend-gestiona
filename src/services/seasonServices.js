const url = "http://localhost:3000/api/v1/seasons"

//GET ALL SEASON
export async function fetchSeasons() {
  const response = await fetch(url,{
    method: "GET",
    headers: {
      "Content-Type": "application/json"
    }
  })
  const data = await response.json()
  return data
}

//POST SEASON
export async function createSeason(season) {
  if (!season || typeof season !== "object")
    throw new Error("The SEASON must be an object");
  const response = await fetch(url,{
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(season)
  })
  const data = await response.json()
  return data
}

//PUT SEASON
export async function editSeason(id,season) {
  if(!id || typeof id !== "string")
    throw new Error("The ID must be an string");
  if(!season || typeof season !== "object")
    throw new Error("The SEASON must be an object");
  const response = await fetch(url+"/"+id,{
    method:"PUT",
    headers: {
      "Content-Type": "application/json"
    },
    body:JSON.stringify(season)
  })
  const data = response.json()
  return data
}

//DELETE SEASON
export async function deleteSeason(id){
  if(!id || typeof id !== "string")
    throw new Error("The ID must be an string");
  const response = await fetch(url+"/"+id,{
    method:"DELETE",
    headers:{
      "Content-Type": "application/json"
    }
  })
  const data = response.json()
  return data
}

