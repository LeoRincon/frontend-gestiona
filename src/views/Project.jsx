import InitialSidebar from './components/InitialSidebar';
import InitialView from './components/InitialView';
import AddProjectModal from './components/AddProjectModal';
import ProjectsList from './components/ProjectsList';
import { useEffect, useRef, useState } from 'react';
import { createProject } from '../services/projectService';

import './styles/project.css';
import { fetchProjects } from '../services/projectService';

export default function Project() {
 const dialogRef = useRef(null);
 const [projects, setProjects] = useState([]);

 useEffect(() => {
  const dataSessionStorage = sessionStorage.getItem('user_data');
  console.log(
   'ðŸ‡¨ðŸ‡´ðŸš¨ => useEffect => dataSessionStorage:',
   JSON.parse(dataSessionStorage)
  );

  const getProjects = async () => {
   const projects = await fetchProjects();
   setProjects(projects);
  };
  getProjects();
 }, []);

 const handleOpenModal = () => {
  if (dialogRef.current) dialogRef.current.showModal();
 };

 const handleCloseModal = () => {
  if (dialogRef.current) dialogRef.current.close();
 };
 const handelAddElement = async (element) => {
  try {
   const response = await createProject(element);
   if (!response) throw new Error('Error al crear el proyecto');
   setProjects((prevProjects) => [...prevProjects, response]);
  } catch (error) {
   console.error(error);
  }
 };

 return (
  <div className='container-project'>
   <InitialSidebar />
   {!projects.length ? (
    <InitialView buttonOnClick={handleOpenModal} />
   ) : (
    <ProjectsList buttonOnClick={handleOpenModal} projects={projects} />
   )}
   <AddProjectModal
    ref={dialogRef}
    onClose={handleCloseModal}
    onAddElement={handelAddElement}
   />
  </div>
 );
}
