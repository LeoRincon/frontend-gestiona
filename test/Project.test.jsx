import { render, screen, waitFor } from "@testing-library/react";
import {
  afterEach,
  beforeAll,
  beforeEach,
  describe,
  expect,
  it,
  vi,
} from "vitest";
import { WrapperTest } from "./WrapperTest.jsx";
import Project from "../src/views/Project.jsx";
import * as projectService from "../src/services/projectService"

vi.mock("../src/services/projectService", () => ({
  getProjectsByUserId: vi.fn(),
  updateSessionProjects: vi.fn(),
}));

describe("Project view", () => {
  beforeAll(() => {
    //Polyfill to simulate the sessionStorage in the test environment
    const storageMock = (() => {
      let store = {};
      return {
        getItem: (key) => store[key] || null,
        setItem: (key, value) => {
          store[key] = String(value);
        },
        removeItem: (key) => {
          delete store[key];
        },
        clear: () => {
          store = {};
        },
      };
    })();

    Object.defineProperty(global, "sessionStorage", {
      value: storageMock,
    });
  });

  beforeEach(() => {
    const mockUserData = JSON.stringify({
      id: 123,
      nombre: "John Doe",
      email: "john.doe@cor.com",
    });

    vi.spyOn(sessionStorage, "getItem").mockImplementation((key) => {
      if (key === "user_data") {
        return mockUserData;
      }
      return null;
    });
  });

  afterEach(() => {
    vi.resetAllMocks(); //restore all mocks to their original implementation
  });

  it("Renderiza InitialView cuando no hay proyectos", () => {
    const { container } = render(
      <WrapperTest>
        <Project />
      </WrapperTest>
    );

    expect(
      screen.getByText(/Crea un nuevo proyecto para comenzar/i)
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Renderiza ProjectsList cuando hay proyectos", async () => {
    const mockProjects = [
      { id: 1, nombre: "Proyecto 1", descripcion: "Descripción 1" },
      { id: 2, nombre: "Proyecto 2", descripcion: "Descripción 2" },
    ];

    projectService.getProjectsByUserId.mockResolvedValue(mockProjects);

    render(
      <WrapperTest>
        <Project />
      </WrapperTest>
    );
    await waitFor(() => {
      expect(screen.getByText(/Proyecto 1/i)).toBeInTheDocument();
      expect(screen.getByText(/Proyecto 2/i)).toBeInTheDocument();
    });
  });
});
