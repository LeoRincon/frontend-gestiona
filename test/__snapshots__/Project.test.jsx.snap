// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Project view > Renderiza InitialView cuando no hay proyectos 1`] = `
<div>
  <div
    class="container-project"
  >
    <aside
      class="sidebar"
    >
      <header
        class="sidebar__header"
      >
        <a
          class=""
          data-discover="true"
          href="/home"
        >
          <svg
            fill="#008164"
            height="60"
            viewBox="0 0 14 16"
            width="52"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.792 5.246a4.008 4.008 0 0 1-.881.848 2.347 2.347 0 0 0 .465-1.35 4.676 4.676 0 0 0-.437-.404 12.463 12.463 0 0 1-2.123 5.273c-.253.356-.512.682-.772.98a2.094 2.094 0 0 1 1.274-.31 2.115 2.115 0 0 1 1.33.64c-.117.142-.788.926-1.916.993a2.686 2.686 0 0 1-1.573-.41c-.34.317-.668.586-.967.813a4.692 4.692 0 0 0 5.6-7.072Zm-1.091 1.56a2.104 2.104 0 0 1 1.09-.544c.014.086.106.706-.336 1.214-.528.608-1.325.49-1.371.482A2.11 2.11 0 0 1 9.7 6.806Zm.293 2.932c-.758.406-1.493.031-1.55 0 .039-.122.263-.775.963-1.13a1.969 1.969 0 0 1 1.443-.123c-.009.073-.118.859-.856 1.253Z"
            />
            <path
              d="M3.792 9.696c.044-.987.546-1.66.734-1.888.23.38.945.922 1.026 2.14a3.24 3.24 0 0 1-.476 1.9 11.05 11.05 0 0 0 1.817-1.691c.481-.561 1.368-1.616 1.928-3.267a9.237 9.237 0 0 0 .485-2.984 4.72 4.72 0 0 0-.613-.295 4.574 4.574 0 0 1-.04 2.316 4.028 4.028 0 0 0-.884-2.572 4.692 4.692 0 0 0-3.043 8.741c-.152-.156-.993-1.05-.933-2.4Zm3.101-5.361c.252.21.868.394 1.257 1.089.534.953.084 1.969.084 2.178-.129-.044-.958-.48-1.34-1.424a2.534 2.534 0 0 1 0-1.843Zm-1.215 3.56c.088-.484.294-.916.46-1.13.61.553.704.607.88 1.34.276 1.138-.517 2.02-.586 2.094-.419-.922-.977-1.076-.754-2.304Z"
            />
            <path
              d="M6.953 0 0 3.984v8.02L6.953 16l6.869-4.063V3.853L6.952 0Zm5.825 11.328-5.831 3.449-5.903-3.392V4.576l5.903-3.382 5.831 3.271v6.863Z"
            />
          </svg>
          <div
            class="header__slogan"
          >
            <h1
              class="slogan__title"
            >
              GestionA
            </h1>
            <h2
              class="slogan__description"
            >
              Sistema de gestión agrícola
            </h2>
          </div>
        </a>
      </header>
      <nav
        class="sidebar__account"
      >
        <p
          class="sidebar__label"
        >
          Cuenta
        </p>
        <ul
          class="sidebar__list"
        >
          <li>
            <a
              class="sidebar__option option__link"
              data-discover="true"
              href="/profile"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.554 1.15a.85.85 0 0 0-.83.662l-.237 1.05c-.38.11-.74.262-1.08.448l-.91-.574a.85.85 0 0 0-1.055.118l-.606.606a.85.85 0 0 0-.118 1.054l.574.912c-.186.338-.337.7-.447 1.079l-1.05.238a.85.85 0 0 0-.662.829v.857a.85.85 0 0 0 .662.829l1.05.238c.11.379.261.74.448 1.08l-.575.91a.85.85 0 0 0 .118 1.055l.606.606a.85.85 0 0 0 1.055.118l.91-.574c.34.186.7.337 1.08.447l.238 1.05a.85.85 0 0 0 .829.662h.857a.85.85 0 0 0 .829-.662l.238-1.05c.379-.11.74-.26 1.08-.447l.91.574a.85.85 0 0 0 1.055-.118l.606-.606a.85.85 0 0 0 .118-1.054l-.574-.911c.187-.34.338-.7.448-1.08l1.05-.238a.85.85 0 0 0 .662-.83v-.856a.85.85 0 0 0-.662-.83l-1.05-.237c-.11-.38-.261-.74-.448-1.08l.575-.911a.85.85 0 0 0-.118-1.054l-.607-.606a.85.85 0 0 0-1.054-.118l-.91.574a5.321 5.321 0 0 0-1.081-.448l-.238-1.05a.85.85 0 0 0-.83-.662h-.856Zm-2.15 3.162a4.476 4.476 0 0 1 1.794-.744L7.554 2h.857l.355 1.568c.66.116 1.27.375 1.796.744l1.36-.857.607.606-.858 1.36c.37.527.628 1.136.744 1.795l1.568.356v.857l-1.568.355a4.476 4.476 0 0 1-.744 1.796l.857 1.36-.606.606-1.36-.857a4.476 4.476 0 0 1-1.796.743L8.411 14h-.857l-.356-1.568a4.475 4.475 0 0 1-1.794-.744l-1.36.858-.607-.606.858-1.36a4.476 4.476 0 0 1-.744-1.796L1.983 8.43v-.857l1.568-.356a4.475 4.475 0 0 1 .743-1.794L3.437 4.06l.606-.606 1.36.857ZM9.507 8a1.525 1.525 0 1 1-3.05 0 1.525 1.525 0 0 1 3.05 0Zm.9 0a2.425 2.425 0 1 1-4.85 0 2.425 2.425 0 0 1 4.85 0Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="option__link--text"
              >
                Configuración
              </span>
            </a>
          </li>
          <li>
            <button
              class="logout-button"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 15 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 1.5a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h7.5a.5.5 0 0 0 0-1H3v-11h7.5a.5.5 0 0 0 0-1H3Zm9.604 3.896a.5.5 0 0 0-.708.708L13.293 7.5H6.5a.5.5 0 0 0 0 1h6.793l-1.397 1.396a.5.5 0 0 0 .708.708l2.25-2.25a.5.5 0 0 0 0-.708l-2.25-2.25Z"
                  fill="#008164"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="logout-button__text"
              >
                Cerrar Sesión
              </span>
            </button>
          </li>
        </ul>
      </nav>
      <footer
        class="sidebar__footer"
      >
        <div
          class="sidebar__profile-image"
        >
          <span>
            J
          </span>
        </div>
        <div
          class="sidebar_profile-information"
        >
          <p
            class="profile__username"
          >
            John Doe
          </p>
          <p
            class="profile__email"
          >
            john.doe@cor.com
          </p>
        </div>
      </footer>
    </aside>
    <main
      class="container__iv"
    >
      <h1
        class="container__iv__title"
      >
        Bienvenido a GestionA
      </h1>
      <div>
        <p
          class="container__iv__description"
        >
          Crea un nuevo proyecto para comenzar.
        </p>
        <p
          class="container__iv__description"
        >
           Haz clic en el siguiente botón para crear un nuevo proyecto
        </p>
      </div>
      <button
        class="btnAdd"
      >
        <svg
          fill="#008164"
          height="50"
          viewBox="0 0 15 15"
          width="50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.49991 0.876892C3.84222 0.876892 0.877075 3.84204 0.877075 7.49972C0.877075 11.1574 3.84222 14.1226 7.49991 14.1226C11.1576 14.1226 14.1227 11.1574 14.1227 7.49972C14.1227 3.84204 11.1576 0.876892 7.49991 0.876892ZM1.82707 7.49972C1.82707 4.36671 4.36689 1.82689 7.49991 1.82689C10.6329 1.82689 13.1727 4.36671 13.1727 7.49972C13.1727 10.6327 10.6329 13.1726 7.49991 13.1726C4.36689 13.1726 1.82707 10.6327 1.82707 7.49972ZM7.50003 4C7.77617 4 8.00003 4.22386 8.00003 4.5V7H10.5C10.7762 7 11 7.22386 11 7.5C11 7.77614 10.7762 8 10.5 8H8.00003V10.5C8.00003 10.7761 7.77617 11 7.50003 11C7.22389 11 7.00003 10.7761 7.00003 10.5V8H4.50003C4.22389 8 4.00003 7.77614 4.00003 7.5C4.00003 7.22386 4.22389 7 4.50003 7H7.00003V4.5C7.00003 4.22386 7.22389 4 7.50003 4Z"
          />
        </svg>
      </button>
    </main>
    <dialog
      class="project-modal"
    >
      <button
        class="close-button btn-close"
      >
        <svg
          fill="#8b8d98"
          height="20"
          viewBox="0 0 15 15"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <section
        class="project-modal__container"
      >
        <h2>
          Crear un nuevo proyecto
        </h2>
        <form
          class="project-modal__form"
        >
          <label
            class="project-modal__form__label"
            for="nombre"
          >
            Nombre Proyecto
          </label>
          <input
            class="project-modal__form__input"
            id="project_name"
            name="nombre"
            placeholder="Escriba el nombre del proyecto"
            required=""
            type="text"
          />
          <label
            class="project-modal__form__label"
            for="descripcion"
          >
            Descripción del proyecto
          </label>
          <textarea
            class="project-modal__form__input"
            id="project_name"
            name="descripcion"
            placeholder="Escriba la descripción del proyecto aquí..."
            required=""
            rows="4"
            type="text"
          />
          <div
            class="project-modal__form__buttons"
          >
            <button
              class="primary-button__component undefined"
              type="Submit"
            >
               
              Crear Proyecto
               
            </button>
            <button
              class="secondary-button"
            >
              Cancelar
            </button>
          </div>
        </form>
      </section>
    </dialog>
  </div>
</div>
`;
